src/compiler/emitter.ts

```diff
@@ -4093,6 +4167,8 @@ module ts {
                         return write("true");
                     case SyntaxKind.FalseKeyword:
                         return write("false");
+                    case SyntaxKind.AsteriskToken:

```

Comment:
There is a precedent, see the `EmptyStatement` case further down in the switch body.

From: Daniel Rosenwasser [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:34 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/emitter.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24627527>:

> @@ -4093,6 +4167,8 @@ module ts {

>                          return write("true");

>                      case SyntaxKind.FalseKeyword:

>                          return write("false");

> +                    case SyntaxKind.AsteriskToken:

This is kind of weird - this is the only case in this switch where we handle non-keyword tokens.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24627527>.